packages/hawtio/src/plugins/camel/CamelContent.tsx

```diff
@@ -66,122 +144,42 @@ export const CamelContent: React.FunctionComponent = () => {
-    { id: 'routes', title: 'Routes', component: <CamelRoutes />, isApplicable: camelService.isRoutesFolder },
-    { id: 'endpoints', title: 'Endpoints', component: <Endpoints />, isApplicable: camelService.isEndpointsFolder },
-    {
-      id: 'routeDiagram',
-      title: 'Route Diagram',
-      component: (
-        <RouteDiagramContext.Provider value={routeDiagramContext}>
-          <RouteDiagram />
-        </RouteDiagramContext.Provider>
-      ),
-      isApplicable: camelService.canViewRouteDiagram,
-    },
-    {
-      id: 'source',
-      title: 'Source',
-      component: <Source />,
-      isApplicable: camelService.canViewSource,
-    },
-    { id: 'properties', title: 'Properties', component: <Properties />, isApplicable: camelService.hasProperties },
-    {
-      id: 'send',
-      title: 'Send',
-      component: <SendMessage />,
-      isApplicable: camelService.canSendMessage,
-    },
-    {
-      id: 'browse',
-      title: 'Browse',
-      component: <BrowseMessages />,
-      isApplicable: camelService.canBrowseMessages,
-    },
-    {
-      id: 'endpoint-stats',
-      title: 'Endpoints (in/out)',
-      component: <EndpointStats />,
-      isApplicable: camelService.canViewEndpointStats,
-    },
-    { id: 'exchanges', title: 'Exchanges', component: <Exchanges />, isApplicable: camelService.hasExchange },
-    {
-      id: 'rest-services',
-      title: 'Rest Services',
-      component: <RestServices />,
-      isApplicable: camelService.hasRestServices,
-    },
-    {
-      id: 'type-converters',
-      title: 'Type Converters',
-      component: <TypeConverters />,
-      isApplicable: camelService.hasTypeConverter,
-    },
-    { id: 'chart', title: 'Chart', component: <Chart />, isApplicable: isDefaultApplicable },
-    // Applicable for same criteria as trace
-    { id: 'profile', title: 'Profile', component: <Profile />, isApplicable: camelService.canTrace },
-    { id: 'trace', title: 'Trace', component: <Trace />, isApplicable: camelService.canTrace },
-    { id: 'debug', title: 'Debug', component: <Debug />, isApplicable: camelService.canGetBreakpoints },
-  ]
-
-  /* Filter the nav items to those applicable to the selected node */
-  const navItems = allNavItems.filter(nav => nav.isApplicable(selectedNode))
-
-  const camelNav = (
-    <Nav aria-label='Camel Nav' variant='tertiary'>
-      <NavList>
-        {navItems.map(nav => (
-          <NavItem key={nav.id} isActive={pathname === `${pluginPath}/${nav.id}`}>
-            <NavLink to={{ pathname: nav.id, search }}>{nav.title}</NavLink>
-          </NavItem>
-        ))}
-      </NavList>
-    </Nav>
+  const camelTabs = (
+    <Tabs mountOnEnter unmountOnExit activeKey={activeTab} onSelect={(_, eventKey) => setActiveTab(eventKey as string)}>
+      {tabItems.map(nav => (
+        <Tab eventKey={nav.id} key={nav.id} title={nav.title}>
+          {camelService.canViewRouteDiagram(selectedNode) ? (
```

Comment:
why do we put this here? it makes the code a bit more cluttered.

Reply:
for passing the Context. I moved the NavItems outside of the component.